src/CreatorToken.sol

```diff
@@ -64,6 +111,35 @@ contract CreatorToken is ERC721 {
     payToken.safeTransferFrom(msg.sender, admin, _adminFee);
   }
 
+  function _sell(uint256 _tokenId, uint256 _minAcceptedPrice) internal whenNotPaused {
+    if (msg.sender != ownerOf(_tokenId)) {
+      revert CreatorToken__CallerIsNotOwner(_tokenId, ownerOf(_tokenId), msg.sender);
+    }
+    if (circulatingSupply == 1) revert CreatorToken__LastTokenCannotBeSold(circulatingSupply);
+
+    uint256 _tokenPrice = _temporaryGetNextTokenPrice();
+    (uint256 _creatorFee, uint256 _adminFee) = calculateFees(_tokenPrice);
+    uint256 _totalPrice = _tokenPrice - _creatorFee - _adminFee;
+
+    if (_totalPrice < _minAcceptedPrice) {
+      revert CreatorToken__MinAcceptedPriceExceeded(_totalPrice, _minAcceptedPrice);
+    }
+
+    transferFrom(msg.sender, address(this), _tokenId);
+    _burn(_tokenId);
+    emit Sold(msg.sender, _tokenId, _tokenPrice, _creatorFee, _adminFee);
+
+    payToken.safeTransfer(creator, _creatorFee);
+    payToken.safeTransfer(msg.sender, _totalPrice);
```

Comment:
This looks like a bug, right? sender should get `_tokenPrice`, not `_totalPrice`. Why don't our tests catch it?

Reply:
I think the confusion is around the naming. 

I'm assuming that `_temporaryGetNextTokenPrice()` will always return `_tokenPrice`

When buying the `_totalPrice` the user pays is `_tokenPrice + _creatorFee + _adminFee;`
When selling the `_totalPrice` that the user gets is `_tokenPrice - _creatorFee - _adminFee;`

Let me know if the assumption is wrong or the naming still doesn't make sense.